### PR TITLE
CLOUDP-73536: Improve cluster create and update to handle reuse of describe output

### PR DESCRIPTION
**Pull request by gssbzn**
## Proposed changes

_Jira ticket:_ CLOUDP-73536

## Checklist

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

File: internal/cli/atlas/clusters/create.go [hunk 1]
```diff
@@ -84,6 +84,16 @@ func (opts *CreateOpts) newCluster() (*atlas.Cluster, error) {
 		if err := file.Load(opts.fs, opts.filename, cluster); err != nil {
 			return nil, err
 		}
+		// There can only be one
```
Comment (gssbzn) [inline internal/cli/atlas/clusters/create.go:87]: update was already doing this so I'm bringing it now

File: examples/atlas/rename-cluster.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
+# Renaming a Cluster
+
+Currently, you're not able to easily rename a cluster, 
+however you could deploy a new cluster with the same properties and only change the name.
```
Comment (andreaangiolillo) [inline examples/atlas/rename-cluster.md:4]: [q] does the new cluster contain the data stored in the old cluster? I think yes but I am not sure.

File: internal/cli/atlas/clusters/create.go [hunk 3]
```diff
@@ -84,6 +84,16 @@ func (opts *CreateOpts) newCluster() (*atlas.Cluster, error) {
...
+		if cluster.ReplicationSpecs != nil {
+			cluster.ReplicationSpec = nil
+		}
+		// This can't be sent
+		cluster.MongoURI = ""
+		cluster.MongoURIWithOptions = ""
+		cluster.MongoURIUpdated = ""
+		cluster.StateName = ""
+		cluster.MongoDBVersion = ""
```
Comment (andreaangiolillo) [inline internal/cli/atlas/clusters/create.go:96]: [q] why are we setting these as empty? Can't we just not set them?

File: internal/cli/atlas/clusters/describe.go [hunk 4]
```diff
@@ -41,11 +41,15 @@ var describeTemplate = `ID	NAME	MDB VER	STATE
...
 func (opts *DescribeOpts) Run() error {
-	r, err := opts.store.Cluster(opts.ConfigProjectID(), opts.name)
+	r, err := opts.store.AtlasCluster(opts.ConfigProjectID(), opts.name)
 	if err != nil {
 		return err
 	}
-
+	// When both are set we can't reuse this output as is as they are both exclusive,
+	// we pick to show the supported property over the deprecated one for this reason
+	if r.ReplicationSpec != nil && r.ReplicationSpecs != nil {
```
Comment (andreaangiolillo) [inline internal/cli/atlas/clusters/describe.go:50]: [q] what's the difference between `ReplicationSpec` and `ReplicationSpecs`?

File: internal/store/clusters.go [hunk 5]
```diff
@@ -126,12 +127,20 @@ func (s *Store) ProjectClusters(projectID string, opts *atlas.ListOptions) (inte
 	}
 }
 
-// Cluster encapsulate the logic to manage different cloud providers
-func (s *Store) Cluster(projectID, name string) (interface{}, error) {
+// AtlasCluster encapsulate the logic to manage different cloud providers
```
Comment (andreaangiolillo) [inline internal/store/clusters.go:130]:
```suggestion
// AtlasCluster encapsulates the logic to manage different cloud providers
```

File: internal/store/clusters.go [hunk 6]
```diff
@@ -126,12 +127,20 @@ func (s *Store) ProjectClusters(projectID string, opts *atlas.ListOptions) (inte
...
 	switch s.service {
 	case config.CloudService:
 		result, _, err := s.client.(*atlas.Client).Clusters.Get(context.Background(), projectID, name)
 		return result, err
+	default:
+		return nil, fmt.Errorf("unsupported service: %s", s.service)
+	}
+}
+
+// OpsManagerCluster encapsulate the logic to manage different cloud providers
```
Comment (andreaangiolillo) [inline internal/store/clusters.go:141]:
```suggestion
// OpsManagerCluster encapsulates the logic to manage different cloud providers
```

**Review by andreaangiolillo — APPROVED** [review]
LGTM, I have just some questions.

File: internal/cli/atlas/clusters/create.go (hunk 3 above)
Comment (gssbzn) [reply internal/cli/atlas/clusters/create.go:96]: when you do describe they are set, if you try to reuse what was output by describe you'll get errors of them being set

File: internal/cli/atlas/clusters/describe.go (hunk 4 above)
Comment (gssbzn) [reply internal/cli/atlas/clusters/describe.go:50]: `ReplicationSpec` is deprecated

**Review by andreaangiolillo — DISMISSED** [review]
LGTM. Thanks for the clarifications

**Review by andreaangiolillo — APPROVED** [review]
LGTM

**Pull request merged**